v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/shard/CassandraShard.java

```diff
@@ -55,35 +54,39 @@ private void extractAndSetHostAndPort() {
     setPort(port);
   }
 
-  private DriverExecutionProfile getProfile() {
-    return configLoader.getInitialConfig().getDefaultProfile();
+  private String getOptionValue(TypedDriverOption<String> key) {
+    return optionsMap.get(key);
   }
 
-  // Getters that fetch data from DriverConfigLoader
-  public List<String> getContactPoints() {
-    return getProfile().getStringList(DefaultDriverOption.CONTACT_POINTS);
+  private List<String> getOptionValueList(TypedDriverOption<List<String>> key) {
+    return optionsMap.get(key);
   }
 
-  public String getKeySpaceName() {
-    return getProfile().getString(DefaultDriverOption.SESSION_KEYSPACE);
+  // Getters that fetch data from OptionsMap
+  public List<String> getContactPoints() {
+    return getOptionValueList(TypedDriverOption.CONTACT_POINTS);
   }
 
-  public DriverConfigLoader getConfigLoader() {
-    return configLoader;
+  public String getKeySpaceName() {
+    return getOptionValue(TypedDriverOption.SESSION_KEYSPACE);
   }
 
   public String getUsername() {
-    return getProfile().getString(DefaultDriverOption.AUTH_PROVIDER_USER_NAME);
+    return getOptionValue(TypedDriverOption.AUTH_PROVIDER_USER_NAME);
   }
 
   public String getPassword() {
```

Comment:
Could you please clarify on where are we using the `getPassword` and `getUsername` getters?

Reply:
we are using getUsername to generate the connection URL

Reply:
If I understand correctly, all these extra getters are used to create a lookup key into a hash map of shard vs session.

Is it must to put the password (and username) as a part of the hash key?  Since Cassandra is a single sharded migration, there's not a big concern on uniqueness of the URL. Even if that was not the case, I don't see password as a must have to guarantee uniqueness.

We should be extra careful about getting out user's credentials and using them at places where it's not strictly necessary. Any future changes that might log the details from the session map might lead to password or username getting logged .

Reply:
@pawankashyapollion where is `getPassword` getting used?

Reply:
@shreyakhajanchi In RR we are not using getPassword anywhere, I can remove it I have added for consistancy